aten/src/ATen/native/TensorShape.cpp

```diff
@@ -153,6 +153,28 @@ Tensor &as_strided_(Tensor& self, IntList size, IntList stride) {
   return at::as_strided_(self, size, stride, self.storage_offset());
 }
 
+Tensor narrow_copy(const Tensor& self, int64_t dim, int64_t start, int64_t length){
+  if (self.type().is_sparse()) {
+    AT_CHECK(self.dim() > 0, "narrow() cannot be applied to a 0-dim tensor.");
+    LongTensor indices = self._indices();
+    int64_t nnz = indices.size(1);
+    int64_t dims = indices.size(0);
+    int64_t end = start+length;
+    
+    std::vector<int64_t> newSizes = self.sizes().vec();
+    newSizes[dim]=length;
+    
+    Tensor mask = (indices[dim] >= start).__and__((indices[dim] < end));
```

Comment:
This isn't right if you have `dim > self.sparseDim()`.

Reply:
Got it, adding my pass at logic as well as a test for that case.